### PR TITLE
Fix hotbar turning dark in third-person when Project Red is installed

### PR DESCRIPTION
**Pull request by embeddedt**
This fixes an annoying issue which causes the hotbar to darken when the player switches to third person mode. I believe the root cause is that lighting is re-enabled in RenderWorldLastEvent by MrTJPCore, when the HUD renderer expects it to be disabled. The workaround is to keep lighting disabled, which is what is done here.

Can someone confirm that they see the issue without the fix applied?

Before:

before

After:

after

Offending code in MrTJPCore: https://github.com/MrTJP/MrTJPCore/blob/adbfa1fe6eff596250949ff13ee3195218db3292/src/mrtjp/core/fx/FXEngine.scala#L140-L144

### CODE REVIEW

**Comment by Glease** [comment]
Are you sure it should be kept disabled? NEI's WorldOverlayRenderer is explicitly turning GL_LIGHTING back on as part of its clean up routine in both renderChunkBounds and renderMobSpawnOverlay, which are all called in codechicken.nei.ClientHandler#renderLastEvent, also an event handler for that event you mentioned.

I can see quite a few people just using glPushAttrib (e.g. ae2) and have 0 idea on what is the actual initial state.

**Comment by embeddedt** [comment]
There's a good chance that 1.7 should be disabling it again somewhere in vanilla code and isn't. The number of GL bugs in these older Minecraft versions is quite astounding. What I know is that I ran the mod in isolation, without any other mods except dependencies, and as soon I disabled the `renderParticles` event handler here, the bug went away. That's how I figured out that enabling lighting here was the culprit, or at least appeared to be.

Is injecting `glPushAttrib`/`glPopAttrib` calls at the beginning and end of this handler a preferable solution?

File: dependencies.gradle [hunk 1]
```diff
@@ -3,6 +3,7 @@ dependencies {
     compile("com.github.GTNewHorizons:StructureLib:1.0.15:dev")
     compile("net.industrial-craft:industrialcraft-2:2.2.828-experimental:dev")
     compile("com.github.GTNewHorizons:AppleCore:3.1.9:dev")
+    compile("com.github.GTNewHorizons:ProjectRed:master-SNAPSHOT:dev")
```
Comment (glowredman) [inline dependencies.gradle:6]: Do you really need to pull ProjectRed? Wouldn't `compile("mrtjp:MrTJPCore:1.7.10-1.1.0.33:dev"` suffice?

Comment (mitchej123) [reply dependencies.gradle:6]: No snapshot Masters.  Pick a tag like glow said

**Review by mitchej123 — CHANGES_REQUESTED** [review]
Tag all deps

File: dependencies.gradle (hunk 1 above)
Comment (embeddedt) [reply dependencies.gradle]: Fixed by using MrTJPCore as glow suggested. Regarding the snapshot issue, I used [this gradle script](https://github.com/GTNewHorizons/ProjectBlue/blob/master/dependencies.gradle#L4) as a reference so I don't know if that needs updating.

Comment (boubou19) [reply dependencies.gradle]: basically, by using snaptshots from master, you compile with the latest code availiable on master. That sounds interesting when you don't want to update the deps in the build script often, but it's actually a bad idea, as your builds become unreproductible.

Comment (mitchej123) [reply dependencies.gradle]: We haven't updated all repos, the one you linked is indeed out of date.

**Review by mitchej123 — APPROVED** [review]
Seems reasonable

**Pull request merged**